src/rebar_utils.erl

```diff
@@ -89,13 +95,128 @@ is_arch(ArchRegex) ->
+    case re:run(Arch, Match, [{capture, none}]) of
+        match -> V;
+        _ ->
+            match_wordsize(Arch, Vs)
+    end.
+
+env_wordsize(Wordsize) when Wordsize =:= false;
+                            Wordsize =:= "" ->
+    io:format("REBAR_TARGET_ARCH_WORDSIZE not set, assuming 32\n"),
+    "32";
+env_wordsize(Wordsize) ->
+    try list_to_integer(Wordsize) of
+        16 -> "16";
+        32 -> "32";
+        64 -> "64";
+        _ ->
+            io:format("REBAR_TARGET_ARCH_WORDSIZE bad value: ~p\n", [Wordsize]),
+            "32"
+    catch
+        error:_ ->
+            io:format("REBAR_TARGET_ARCH_WORDSIZE bad value: ~p\n", [Wordsize]),
+            "32"
+    end.
+
+%%
+%% Findout the word size of the target by using Arch-gcc
+%%
+
+cross_wordsize(Arch) ->
+    cross_sizeof(Arch, "void*").
+
+%%
+%% Find the size of target Type using a specially crafted C file 
+%% that will report an error on the line of the byte size of the type.
+%% 
+                       
+cross_sizeof(Arch, Type) ->
+    Compiler = if  Arch =:= "" -> "cc";
+                   true -> Arch ++ "-gcc"
+               end,
+    TempFile = mktemp(".c"),
+    file:write_file(TempFile,
+                    <<"int t01 [1 - 2*(((long) (sizeof (TYPE))) == 1)];\n"
+                      "int t02 [1 - 2*(((long) (sizeof (TYPE))) == 2)];\n"
+                      "int t03 [1 - 2*(((long) (sizeof (TYPE))) == 3)];\n"
+                      "int t04 [1 - 2*(((long) (sizeof (TYPE))) == 4)];\n"
+                      "int t05 [1 - 2*(((long) (sizeof (TYPE))) == 5)];\n"
+                      "int t06 [1 - 2*(((long) (sizeof (TYPE))) == 6)];\n"
+                      "int t07 [1 - 2*(((long) (sizeof (TYPE))) == 7)];\n"
+                      "int t08 [1 - 2*(((long) (sizeof (TYPE))) == 8)];\n"
+                      "int t09 [1 - 2*(((long) (sizeof (TYPE))) == 9)];\n"
+                      "int t10 [1 - 2*(((long) (sizeof (TYPE))) == 10)];\n"
+                      "int t11 [1 - 2*(((long) (sizeof (TYPE))) == 11)];\n"
+                      "int t12 [1 - 2*(((long) (sizeof (TYPE))) == 12)];\n"
+                      "int t13 [1 - 2*(((long) (sizeof (TYPE))) == 13)];\n"
+                      "int t14 [1 - 2*(((long) (sizeof (TYPE))) == 14)];\n"
+                      "int t15 [1 - 2*(((long) (sizeof (TYPE))) == 15)];\n"
+                      "int t16 [1 - 2*(((long) (sizeof (TYPE))) == 16)];\n"
+                    >>),
+    Res = os:cmd(Compiler ++ " -DTYPE=\""++Type++"\" " ++ TempFile),
+    file:delete(TempFile),
+    case string:tokens(Res, ":") of
+        [_,Ln | _] ->
+            try list_to_integer(Ln) of
+                NumBytes -> integer_to_list(NumBytes*8)
+            catch
+                error:_ ->
+                    ""
+            end;
+        _ ->
+            ""
+    end.
+
+mktemp(Suffix) ->
+    {A,B,C} = erlang:now(),
+    Dir = case os:type() of
+              {windows,_} -> "C:/WINDOWS/TEMP";
+              _ -> "/tmp"
```

Comment:
Hmm, isn't there a platform abstraction to get the temp dir. We should at least use the `%TEMP%` env var on Windows.